src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -84,18 +85,29 @@ def evaluate_workflows_action_filters(
 ) -> BaseQuerySet[Action]:
     filtered_action_groups: set[DataConditionGroup] = set()
 
-    # gets the list of the workflow ids, and then get the workflow_data_condition_groups for those workflows
-    workflow_ids = {workflow.id for workflow in workflows}
+    # Gets the list of the workflow ids, and then get the workflow_data_condition_groups for those workflows
+    workflow_ids_to_envs = {workflow.id: workflow.environment for workflow in workflows}
 
     action_conditions = DataConditionGroup.objects.filter(
-        workflowdataconditiongroup__workflow_id__in=workflow_ids
+        workflowdataconditiongroup__workflow_id__in=list(workflow_ids_to_envs.keys())
     ).distinct()
 
+    workflow_to_dcg = dict(
+        WorkflowDataConditionGroup.objects.filter(
+            condition_group_id__in=action_conditions
+        ).values_list("condition_group_id", "workflow_id")
+    )
+
     for action_condition in action_conditions:
-        # TODO(cathy): attach correct workflow to job
+        workflow_job = job
```

Comment:
i need to fix this naming to stop calling these "jobs", will update in a separate PR